osutil/mountentry_linux.go

```diff
@@ -26,6 +26,54 @@ import (
 	"syscall"
 )
 
+// splitMountoptions will split options at ',' accounting for escaped ','
+// characters. ',' can be found as part of a valid path for instance.
+func splitMountOptions(s string) []string {
+	var o []string
+
+	var cur strings.Builder
+	escaped := false
+	for _, c := range s {
+		switch c {
+		case '\\':
+			if escaped {
+				cur.WriteRune(c)
+				escaped = false
+			} else {
+				escaped = true
+			}
+		case ',':
+			if escaped {
+				cur.WriteRune(c)
+				escaped = false
+			} else {
+				o = append(o, cur.String())
+				cur.Reset()
+			}
+		default:
+			if escaped {
+				cur.WriteRune('\\')
+				escaped = false
+			}
+			cur.WriteRune(c)
+		}
+	}
+
+	if cur.Len() > 0 {
+		o = append(o, cur.String())
+	}
+
+	return o
+}
+
+// SplitSystemdMountOptions is used to properly split options passed to systemd-mount
+// that could contain an escaped ',' character such as a verity hash device or overlayfs
+// mount options.
+// Relevant systemd code https://github.com/systemd/systemd/blob/4ec630bfba02543af204f13ad9f5ce4a2329a166/src/basic/extract-word.c#L20
+func SplitSystemdMountOptions(s string) []string {
```

Comment:
Is this used only in tests? Should it be moved there as well then?

Reply:
With my latest commits (that I reverted its usage it in `ParseMountEntry()`), yes. But take a look at https://github.com/canonical/snapd/pull/14179#discussion_r1682692851. The idea with moving it in osutil was so that we gather everything related to mount/options parsing in one place so that's easier to track.

Also parsing options passed to `systemd-mount --options=` turned out to not be exactly the same as parsing fstab options and that's why I reverted the change in the context of this PR.